credentials/google/google.go

```diff
@@ -35,63 +35,56 @@ const tokenRequestTimeout = 30 * time.Second
 
 var logger = grpclog.Component("credentials")
 
-// NewDefaultCredentials returns a credentials bundle that is configured to work
-// with google services.
+// DefaultCredentialsOptions constructs options to build DefaultCredentials.
+type DefaultCredentialsOptions struct {
+	// PerRPCCreds is a  per RPC credentials that is passed to a bundle.
+	PerRPCCreds credentials.PerRPCCredentials
+}
+
+// NewDefaultCredentialsWithOptions returns a credentials bundle that is
+// configured to work with google services.
 //
 // This API is experimental.
-func NewDefaultCredentials() credentials.Bundle {
+func NewDefaultCredentialsWithOptions(opts DefaultCredentialsOptions) credentials.Bundle {
+	perRPC := opts.PerRPCCreds
+	if perRPC == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), tokenRequestTimeout)
+		defer cancel()
+		var err error
+		perRPC, err = oauth.NewApplicationDefault(ctx)
+		if err != nil {
+			logger.Warningf("google default creds: failed to create application oauth: %v", err)
+		}
+	}
 	c := &creds{
 		newPerRPCCreds: func() credentials.PerRPCCredentials {
-			ctx, cancel := context.WithTimeout(context.Background(), tokenRequestTimeout)
-			defer cancel()
-			perRPCCreds, err := oauth.NewApplicationDefault(ctx)
-			if err != nil {
-				logger.Warningf("google default creds: failed to create application oauth: %v", err)
-			}
-			return perRPCCreds
+			return perRPC
 		},
 	}
 	bundle, err := c.NewWithMode(internal.CredsBundleModeFallback)
 	if err != nil {
-		logger.Warningf("google default creds: failed to create new creds: %v", err)
+		logger.Warningf("google default creds with per rpc: failed to create new creds: %v", err)
```

Comment:
"with per rpc"?  Leave it at "google default creds" or use the function name directly?  Note this doesn't match the logging line above.

Reply:
Changed to function name.